meta/AccessTableLookup.php

```diff
@@ -88,27 +69,31 @@ public function saveData($data) {
                 $opt[] = $value;
             }
         }
-        $singlesql = "REPLACE INTO $stable (" . join(',', $singlecols) . ") VALUES (" . trim(str_repeat('?,', count($opt)), ',') . ")";
+
+        $ridSingle = "(SELECT (COALESCE(MAX(rid), 0 ) + 1) FROM $stable)";
+        $ridMulti = "(SELECT (COALESCE(MAX(rid), 0 ) + 1) FROM $mtable)";
+
+        $singlesql = "REPLACE INTO $stable (pid, rid, " . join(',', $singlecols) . ") VALUES (NULL, $ridSingle, " . trim(str_repeat('?,', count($opt)), ',') . ")";
         /** @noinspection SqlResolve */
-        $multisql = "REPLACE INTO $mtable (pid, colref, row, value) VALUES (?,?,?,?)";
+        $multisql = "REPLACE INTO $mtable (pid, rid, colref, row, value) VALUES (NULL, $ridMulti, ?,?,?)";
 
         $this->sqlite->query('BEGIN TRANSACTION');
         $ok = true;
 
         // insert single values
         $ok = $ok && $this->sqlite->query($singlesql, $opt);
 
-        // get new pid if this is a new insert
-        if($ok && !$this->pid) {
+        // get new rid if this is a new insert
+        if($ok && !$this->rid) {
             $res = $this->sqlite->query('SELECT last_insert_rowid()');
```

Comment:
Is this still correct? The internal `ROWID` is probably not equal to our `rid`?

Reply:
Right, this must be replaced with a normal SELECT MAX()

Reply:
Hmm inserting with `MAX()+1` and later selecting `MAX()` may not be atomic. A wrapping transaction might make it atomic (but I'm not sure what the isolation level for sqlite transactions is). It might be possible to use `SELECT rid FROM <table> WHERE ROWID = last_insert_rowid()`.